test/rest_api/test_column_family.py

```diff
@@ -85,15 +85,23 @@ def test_column_family_tombstone_gc_correctness(cql, this_dc, rest_api):
             # 3 Create partition tombstone
             cql.execute(f"DELETE from {test_keyspace}.{test_table} WHERE a = 1")
 
-            # 4 Flush it into sstable
-            resp = rest_api.send("POST", f"storage_service/keyspace_flush/{test_keyspace}")
+            # 4 Trigger major compaction on that sstable (flushes memtable by default)
+            resp = rest_api.send("POST", f"column_family/major_compaction/{test_keyspace}:{test_table}")
```

Comment:
I don't understand something here.... It seems you are assuming that major compaction will flush the memtable. But I see two patches later that there is some 1-day threshold that may avoid this flush. So is this flush guaranteed? Was it always guaranteed, also before this patch, and the flush step was always unnecessary?

Reply:
The addition in this series is the flush of **all** tables in the database before major compaction.
Flushing the particular tables that are major compacted was already done before (in all supported versions), and is still done in the case that flushing all db tables is skipped due to the 1-day threshold (but not if the `flush_memtables=false` option is given).